integration/internal/process_test.go

```diff
@@ -28,45 +31,155 @@ var _ = Describe("Start method", func() {
 		Consistently(processState.Session.ExitCode).Should(BeNumerically("==", -1))
 	})
 
-	Context("when process takes too long to start", func() {
-		It("returns a timeout error", func() {
-			processState := &ProcessState{}
-			processState.Path = "bash"
-			processState.Args = simpleBashScript
-			processState.StartTimeout = 200 * time.Millisecond
-			processState.StartMessage = "loop 5000"
+	Context("when a health check endpoint is provided", func() {
+		var server *ghttp.Server
+		BeforeEach(func() {
+			server = ghttp.NewServer()
 
-			err := processState.Start(nil, nil)
-			Expect(err).To(MatchError(ContainSubstring("timeout")))
+		})
+		AfterEach(func() {
+			server.Close()
+		})
 
-			Eventually(processState.Session.ExitCode).Should(Equal(143))
+		Context("when the healthcheck returns ok", func() {
+			BeforeEach(func() {
+				server.RouteToHandler("GET", "/healthz", ghttp.RespondWith(http.StatusOK, ""))
+			})
+
+			It("hits the endpoint, and successfully starts", func() {
+				processState := &ProcessState{}
```

Comment:
DRY - Consider creating a function to initialize a ``ProcessState`` instance rather than duplicating those details in every test.

Reply:
We **try** to keep the tests as easy to grasp as possible and to have them tell a story. That means that we sometimes/often deliberately end up with rather verbose tests and duplication in them.

Having said that, I am happy to revisit this. Especially, but not only, in this case it might be better to abstract all that into a `func createSimpleProcessState() *ProcessState` or such.